src/consumption.cpp

```diff
@@ -1024,14 +1024,16 @@ void Character::modify_health( const islot_comestible &comest )
 
 void Character::modify_stimulation( const islot_comestible &comest )
 {
+    if( comest.stim == 0 ) {
+        return;
+    }
     const int current_stim = get_stim();
-    if( comest.stim != 0 &&
-        ( std::abs( current_stim ) < ( std::abs( comest.stim ) * 3 ) ||
-          sgn( current_stim ) != sgn( comest.stim ) ) ) {
-        if( comest.stim < 0 ) {
-            set_stim( std::max( comest.stim * 3, current_stim + comest.stim ) );
+    if( comest.stim != 0 || sgn( current_stim ) != sgn( comest.stim ) ) {
+        if( ( std::abs( comest.stim ) * 3 ) > std::abs( current_stim ) ) {
+            mod_stim( comest.stim );
         } else {
-            set_stim( std::min( comest.stim * 3, current_stim + comest.stim ) );
+            comest.stim > 0 ? mod_stim( std::max( comest.stim / 2, 1 ) ) : mod_stim( std::min( comest.stim / 2,
+                    -1 ) );
```

Comment:
This is always true since if comest.stim == 0 we already returned from the function.

Reply:
Umm... what about if comest.stim is below 0?